42rave/transcendence
# MERGE PRODUCTION ENVIRONMENT:

**Description:**
This pull request introduces the necessary changes to add a production environment configuration to the project.
It includes updates to configuration files and Docker setups to ensure a seamless transition to production mode.

**Changes Made:**

1. **.env.sample**:
   - Added production-specific environment variables.
   - Updated API_URL and WS_BASE_URL to match production URLs.

2. **.gitignore**:
   - Added .env.dev to the gitignore file.

3. **back/Dockerfile**:
   - Added a new Docker build stage for production.
   - Copies the necessary production files and dependencies.
   - Sets NODE_ENV to production for the production image.

4. **back/entrypoint.sh**:
   - Updated to export NODE_ENV as production.

5. **front/Dockerfile**:
   - Added a new Docker build stage for production.
   - Copies the production build output.

6. **front/entrypoint.sh**:
   - Updated to export NODE_ENV as production.

7. **front/nuxt.config.ts**:
   - Updated the default API_URL and WS_BASE_URL for production.

8. **front/stores/auth.store.ts**:
   - Updated API_URL references to use the production URL.

9. **docker-compose.dev.yml**:
   - Added a development-specific Docker Compose file.

10. **docker-compose.yml**:
    - Updated Docker Compose file for production.
    - Added NGINX configuration for reverse proxying to the frontend and backend.

11. **proxy/templates/default.conf.template**:
    - Added NGINX configuration for reverse proxying to the frontend, backend, and WebSocket.

12. **proxy/entrypoint-wrapper.sh**:
    - Added a script to control NGINX start based on NODE_ENV.

**How to Test:**
1. Clone this branch and navigate to the project directory.
2. Run `docker-compose up --build` for production or `docker-compose -f docker-compose.yml -f docker-compose.dev.yml up --build` for development.
3. Verify that the project runs without errors in both development and production environment.

Close #49
